lightwood/encoder/categorical/autoencoder.py

```diff
@@ -10,98 +10,198 @@
+            if type(module) != torch.nn.Sequential and type(module) != DefaultNet
+        ]
+
         self.encoder = torch.nn.Sequential(*modules[0:2]).eval()
         self.decoder = torch.nn.Sequential(*modules[2:3]).eval()
         log.info('Categorical autoencoder ready')
 
-        self.output_size = self.onehot_encoder.output_size
-        self.output_size = self.max_encoded_length
         self.is_prepared = True
 
-    def encode(self, column_data):
+    def encode(self, column_data: Iterable[str]) -> torch.Tensor:
+        """
+        Encodes categorical information in column as the compressed vector from the CatAE.
+
+        :param column_data: An iterable of category samples from a column
+
+        :returns: An embedding for each sample in original input
+        """  # noqa
         oh_encoded_tensor = self.onehot_encoder.encode(column_data)
 
         with torch.no_grad():
             oh_encoded_tensor = oh_encoded_tensor.to(self.net.device)
             embeddings = self.encoder(oh_encoded_tensor)
             return embeddings.to('cpu')
 
-    def decode(self, encoded_data):
+    def decode(self, encoded_data: torch.Tensor) -> List[str]:
+        """
+        Decodes from the embedding space, the original categories.
+
+        ..warning If your reconstruction accuracy is not 100%, the CatAE may not return the correct category.
+
+        :param encoded_data: A torch tensor of embeddings for category predictions
+
+        :returns: A list of 'translated' categories for each embedding
+        """  # noqa
         with torch.no_grad():
             encoded_data = encoded_data.to(self.net.device)
             oh_encoded_tensor = self.decoder(encoded_data)
             oh_encoded_tensor = oh_encoded_tensor.to('cpu')
             return self.onehot_encoder.decode(oh_encoded_tensor)
+
+    def _prepare_AE_input(
+        self, train_priming_data: pd.Series, dev_priming_data: pd.Series
+    ) -> Tuple[DataLoader, DataLoader]:
+        """
+        Creates the data loaders for the CatAE model inputs. Expected inputs are generally of form `pd.Series`
+
+        Currently does not use 'dev'; concatenates both inputs together.
+
+        Input to the `DataLoader` must be an Iterable[str] (ideally List[str])
+
+        """  # noqa
+        if len(dev_priming_data) > 0:
+            priming_data = (
+                pd.concat([train_priming_data, dev_priming_data]).astype(str).tolist()
+            )
+        else:
+            priming_data = [str(x) for x in train_priming_data]
+
+        random.seed(len(priming_data))
+
+        # Prepare a one-hot encoder for CatAE inputs
+        self.onehot_encoder.prepare(priming_data)
+        self.batch_size = min(self.batch_size, int(len(priming_data) / 50))
+
+        train_loader = DataLoader(
+            list(zip(priming_data, priming_data)),
+            batch_size=self.batch_size,
+            shuffle=True,
+        )
+
+        # TODO; make `Gym` compatible with a dev set
```

Comment:
Hmh, will turn this into an issue.

Reply:
Wait, am now a bit confused. Gym supports a `test_data_loader`, which I would think works for this case. @George3d6 thoughts?